bindata/etcd/restore-pod.yaml

```diff
@@ -24,6 +24,43 @@ spec:
         export ETCD_NAME=${NODE_NODE_ENVVAR_NAME_ETCD_NAME}
         export ETCD_INITIAL_CLUSTER="${ETCD_NAME}=https://${NODE_NODE_ENVVAR_NAME_ETCD_DNS_NAME}:2380"
         env | grep ETCD | grep -v NODE
+        export ETCD_NODE_PEER_URL=https://${NODE_NODE_ENVVAR_NAME_ETCD_DNS_NAME}:2380
+
+        # checking if data directory is empty, if not etcdctl restore will fail
+        if [ ! "$(ls -A /var/lib/etcd)" ]; then
+          echo "please delete the data directory before restoring"
+          exit 1
+        fi
+
+        # check if we have backup file to be restored and if the backup file is not currently
+        # being copied, exit non-zero if copying continues for more than 5 seconds
+        if [ ! -f var/lib/etcd-backup/snapshot.db ]; then
+          echo "please make a copy of the backup file, then move that copy to /var/lib/etcd-backup/snapshot.db"
+          exit 1
+        else
+          filesize=$(stat --format=%s "/var/lib/etcd-backup/snapshot.db")
+          for i in `seq 1 5`;
+          do
+            newfilesize=$(stat --format=%s "/var/lib/etcd-backup/snapshot.db")
+            if [ "$filesize" == "$newfilesize" ]; then
+              break
+            fi
+            sleep 1
+          done
+          if [ "$filesize" != "$newfilesize" ]; then
+            echo "file size is changing for more than 5 seconds"
+            exit 1
+          fi
+        fi
+
+        NEW_UUID=$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 10 | head -n 1)
+        echo "restoring to a single node cluster"
+        ETCDCTL_API=3 /usr/bin/etcdctl snapshot restore /var/lib/etcd-backup/snapshot.db \
+         --name  $ETCD_NAME \
+         --initial-cluster=$ETCD_INITIAL_CLUSTER \
+         --initial-cluster-token "openshift-etcd-$NEW_UUID" \
```

Comment:
This needs to be part of restore for all the other members we add how does that work

Reply:
actually we are doing single-member restore but we will need to make sure that add will succeed without this.

Reply:
cleared over slack, we dont need to worry about the token since this is a single member etcd cluster. Other members will be added by clustermembercontroller